### ISSUE TITLE
Liveness Probe For Nat Streaming Server Cluster

### ISSUE THREAD

**Issue by Mr7Sir** [nats-io/nats-streaming-server#843]
Hi,
I need to implement a nats streaming server cluster for fault tolerance on kubernetes but am having trouble implementing a liveness/readiness check for both the server and the client (the client doesn't implement an httpserver). Is there any functionality in nats-streaming server and client that I'm missing which enables these checks for kubernetes?

**Comment by wallyqs** [comment]
There is a monitoring endpoint for the nats-streaming-server when you pass the `-m` flag. You could then send an http request to `/streaming/` for example: https://github.com/nats-io/nats-streaming-server/#monitoring

**Comment by Mr7Sir** [comment]
And what about the NATS client? Is there a way to check if the client is in hung state if the client is in Kubernetes pod

**Comment by ripienaar** [comment]
As you use the client in your own program it's up to your own code how you wish to expose liveness, one would not want a NATS client to just go and open some HTTP port it's not related.

Suggest you implement an app specific liveness for your own needs.

**Comment by kozlovic** [comment]
Agreed with @ripienaar, this is something that you need to include in your application. Suppose that your app uses 10 different libraries, if every library had a liveness probe, that would be a waste..

But know that the streaming library has a ping mechanism with the server so it will report an error if it loses its connection to the streaming server. You can find more information [here](https://github.com/nats-io/stan.go#connection-status).
